File: AAS.BackOffice/client_app/src/domain/users/userBlank.ts
import { User } from "./user"

export interface UserBlank {
    id: string | null,
    firstName: string | null,
    middleName: string | null,
    lastName: string | null,
    email: string | null,
    phoneNumber: string | null,
    roleId: string | null,
    password: string | null,
    rePassword: string | null
}

export namespace UserBlank {
    export function getDefault(): UserBlank {
        return {
            id: null,
            firstName: null,
            middleName: null,
            lastName: null,
            email: null,
            phoneNumber: null,
            roleId: null,
            password: null,
            rePassword: null
        }
    }

    export function fromUser(user: User, roleId: string | null = null): UserBlank {
        return {
            id: user.id,
            firstName: user.firstName,
            middleName: user.middleName,
            lastName: user.lastName,
            email: user.email,
            phoneNumber: user.phoneNumber,
            roleId: roleId,
            password: null,
            rePassword: null
        }
    }
}

File: AAS.Domain/Users/UserBlank.cs
#region

using System.Text.Json.Serialization;
using AAS.Tools.Managers;
using AAS.Tools.Types.IDs;

#endregion

namespace AAS.Domain.Users;

public class UserBlank
{
    public ID? Id { get; set; }
    public string? FirstName { get; set; }
    public string? MiddleName { get; set; }
    public string? LastName { get; set; }
    public string? Email { get; set; }
    public string? PhoneNumber { get; set; }
    public ID? RoleId { get; set; }
    public string? Password { get; set; }
    public string? RePassword { get; set; }

    [JsonIgnore] public string? Passwordhash => HashManager.DefinePasswordHash(Password);

    public UserBlank(ID? id, string? firstName, string? middleName, string? lastName, string? email,
        string? phoneNumber, ID? roleId, string? password, string? rePassword)
    {
        Id = id;
        FirstName = firstName;
        MiddleName = middleName;
        LastName = lastName;
        Email = email;
        PhoneNumber = phoneNumber;
        RoleId = roleId;
        Password = password;
        RePassword = rePassword;
    }
}